x-pack/plugins/rule_registry/server/rule_data_client/index.ts

```diff
@@ -102,15 +103,78 @@ export class RuleDataClient implements IRuleDataClient {
     };
   }
 
-  async createOrUpdateWriteTarget({ namespace }: { namespace?: string }) {
+  async rolloverAliasIfNeeded({ namespace }: { namespace?: string }) {
+    const clusterClient = await this.getClusterClient();
+    const alias = getNamespacedAlias({ alias: this.options.alias, namespace });
+    let simulatedRollover: IndicesRolloverResponse;
+    try {
+      ({ body: simulatedRollover } = await clusterClient.indices.rollover({
+        alias,
+        dry_run: true,
+      }));
+    } catch (err) {
+      if (err?.meta?.body?.error?.type !== 'index_not_found_exception') {
+        throw err;
+      }
+      return;
+    }
+
+    const [writeIndexMapping, simulatedIndexMapping] = await Promise.all([
+      clusterClient.indices.get({
+        index: simulatedRollover.old_index,
+      }),
+      clusterClient.indices.simulateIndexTemplate({
+        name: simulatedRollover.new_index,
+      }),
+    ]);
+    const currentVersions = get(writeIndexMapping, [
+      'body',
+      simulatedRollover.old_index,
+      'mappings',
+      '_meta',
+      'versions',
+    ]);
+    const targetVersions = get(simulatedIndexMapping, [
+      'body',
+      'template',
+      'mappings',
+      '_meta',
+      'versions',
+    ]);
+    const componentTemplateRemoved =
+      Object.keys(currentVersions).length > Object.keys(targetVersions).length;
```

Comment:
This line fails with `TypeError: Cannot convert undefined or null to object` when `currentVersions` is undefined. Let's add an additional guard here.

Reply:
Interesting, how's that possible. Both current and simulated indices should have this `versions` object in `_meta` (I guess?). Maybe this is an indication of a bug here?

Reply:
Maybe it happened when switching from another branch to this one while using the same ES cluster? I'm extracting this logic into a function and making it a little more robust with guards around the cases where currentVersions or targetVersions are undefined.

Reply:
In my case, there was no `_meta` field in the response. Not sure how it happened, tbh.